hashicorp/boundary-ui
# Icu 12781 add filtering pagination support for host catalogs

✅ Closes: https://hashicorp.atlassian.net/browse/ICU-12781

## Description
Adds search support for host catalogs. User can search by id, name, description, scope, plugin type.

No filters are added here as we display in the table the plugin name (aws and azure) as also a "type". It's quite hard to form the correct filters for both the type and plugin name in one filter dropdown. An alternative is to have both a type and a provider dropdown but after [talking about it with design](https://hashicorp.slack.com/archives/C0647GX0TLJ/p1712005802193929), we decided not to include any filters.  

## Screenshots (if appropriate):

https://github.com/hashicorp/boundary-ui/assets/5783847/1a8b789b-cfdc-481d-b0f9-9478f12e2f36


## How to Test

1. Start boundary dev instance
2. Test search on host catalogs
5. Validate searching works as expected

## Checklist:
